### PR TITLE
ADC-voltage conversion in StreamBufferHeader model

### PR DESCRIPTION
**Pull request by t-sasatani**
I'm doing this quick PR because I needed it for hardware experiments. This PR is just for converting ADC to voltage in the `StreamBufferHeader` model using the YAML device config. I think I'll update to `v0.4.1` combined with #41 if this looks quick enough.

### CODE REVIEW

**Comment by sneakers-the-rat** [comment]
Rebased this branch against main after doing the merge of the dummy word commit, just letting ya know so its not confusing if git says your branch is out of sync :)

**Comment by sneakers-the-rat** [comment]
This all makes perfect sense, I have some minor semantic suggestions for handling the scaled values so ill pitch that when I get back to keyboard, but I think we can merge this today

**Review by sneakers-the-rat — COMMENTED** [review]
OK - looks good, i'm going to make two minor changes w/ how the scaling is applied, but then this should be good to go.

Also as i'm writing this i'm realizing that it's more annoying of me to make all these suggestions and make you switch them rather than do a rename myself, so i'll do that too

**Review by sneakers-the-rat — APPROVED** [review]
OK I know that the changes in https://github.com/Aharoni-Lab/miniscope-io/pull/44/commits/d8c01a50c6c5d2ea620b6208091cd499684f0004 might come across as pedantic but they have a point i promise.

- Moving the methods to the scaling object: keeping methods grouped together with the thing that applies them. Adding functionality with composition, encouraging reusability. We will eventually have multiple stream data headers that each want to have their own scaling object, so this sets us up for that. Ideally we would have a `ScaledValue` class that we would apply directly to the model field like `ScaledValue[float, _scaling_fn]` to reduce boilerplate, but didn't want to change too much rn. It also allows us to simplify tests bc the scaling methods are usable outside the header model. 
- `property` vs `computed_value` - `property`s don't show up when the model is dumped, but `computed_field`s are.
- Applying scaling when header is instantiated - keeping similar calls grouped together again: the header should be complete and parsed when it leaves the `_parse_header` method. Since passing the object takes effectively no time, we might as well always do it.
- Returning unscaled values when scaler is absent - we will probably want it to be possible to just return raw values at some point (ie. setting `adc_scale: ` to null in the config), and so we don't want to special-case the accessing of those values by having to get the `battery_voltage_adc` attr instead.

So i'm good to merge now :)

File: miniscope_io/models/stream.py [hunk 1]
```diff
@@ -14,6 +14,29 @@
 from miniscope_io.models.sinks import CSVWriterConfig, StreamPlotterConfig
 
 
+class adcScaling(MiniscopeConfig):
```
[the remark this answers is not in the document]
Comment (t-sasatani) [reply miniscope_io/models/stream.py]: Makes sense. Always feel I should look into this but never done it...

File: miniscope_io/models/stream.py [hunk 2]
```diff
@@ -41,8 +69,43 @@ class StreamBufferHeader(BufferHeader):
     """
 
     pixel_count: int
-    battery_voltage: int
-    ewl_pos: int
+    battery_voltage_adc: int
```
[the remark this answers is not in the document]
Comment (t-sasatani) [reply miniscope_io/models/stream.py:93]: Yeah that makes sense will do

**Comment by t-sasatani** [comment]
Thanks, @sneakers-the-rat. The updates look nice. I was wondering where to put that conversion method and put it into the model because it felt too nested otherwise. But yeah, it's better to have it inside the scaling model, thinking about reuse. Also, I didn't know the `computed_value` field, but it seems better here.

**Pull request merged**